lib/datadog/appsec/configuration/settings.rb

```diff
@@ -53,73 +52,71 @@ def self.add_settings!(base)
               end
 
               option :ruleset do |o|
-                o.default { ENV.fetch('DD_APPSEC_RULES', DEFAULT_APPSEC_RULESET) }
+                o.env_var 'DD_APPSEC_RULES'
+                o.default DEFAULT_APPSEC_RULESET
               end
 
               option :ip_denylist do |o|
-                o.default { [] }
+                o.type :array
+                o.default []
               end
 
               option :user_id_denylist do |o|
-                o.default { [] }
+                o.type :array
+                o.default []
               end
 
               option :waf_timeout do |o|
-                o.default { ENV.fetch('DD_APPSEC_WAF_TIMEOUT', DEFAULT_APPSEC_WAF_TIMEOUT) } # us
+                o.env_var 'DD_APPSEC_WAF_TIMEOUT' # us
+                o.default DEFAULT_APPSEC_WAF_TIMEOUT
                 o.setter do |v|
                   Datadog::Core::Utils::Duration.call(v.to_s, base: :us)
```

Comment:
> in our other libraries

Except in Go, coz it's [parsed as a `time.Duration`](https://github.com/DataDog/dd-trace-go/blob/be1878f26eff1be2fde73bc1819d2e6db792f4f9/internal/appsec/config.go#L115)) using [`time.ParseDuration`](https://pkg.go.dev/time#ParseDuration), which [doesn't accept unitless strings](https://play.golang.com/p/CJfhzjMBmOy), and the env var got used with a suffix in system tests coz Go was the first one there, and then it got in the wild, so I had to implement this parser otherwise both system tests and Go customers having already set env vars in some global way would crash.

But they [fixed since](https://github.com/DataDog/dd-trace-go/blob/be1878f26eff1be2fde73bc1819d2e6db792f4f9/internal/appsec/config.go#L111-L113) by appending `us` when it's missing and IIRC system tests have been adjusted to not use the suffix.

Reply:
> Consider maybe going back to o.type :int in the future?

I'm wondering if an alternative would be to have a `o.type :duration` shorthand and apply that wherever we have durations.